lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1341,13 +1348,39 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
                         chain=certificate["chain"],
                     )
                 else:
+                    if JujuVersion.from_environ().has_secrets:
```

Comment:
`_on_relation_changed` is getting quite complex, I believe extending the docstring to explain what we're doing may be useful.

Reply:
Added, please take a look at it.